src/AccountingModule.sol

```diff
@@ -0,0 +1,173 @@
+     * and burns an equiv amount of accounting tokens
+     * @param amount amount to deposit
+     */
+    function withdraw(uint256 amount) external onlyStrategy {
+        ACCOUNTING_TOKEN.burnFrom(msg.sender, amount);
+        IERC20(BASE_ASSET).safeTransferFrom(safe, msg.sender, amount);
+    }
+
+    /**
+     * @notice Process rewards by minting accounting tokens
+     * @param amount profits to mint
+     */
+    function processRewards(uint256 amount) external onlySafeManager checkAndResetCooldown {
+        uint256 totalSupply = ACCOUNTING_TOKEN.totalSupply();
+
+        // sanity check: if token.totalSupply() > small amount to prevent rounding issues
+        if (totalSupply < 1 ether) revert TvlTooLow();
+
+        // check for upper bound
+        // targetApy / year * token.totalsupply()
+        if (amount > targetApy * totalSupply / DIVISOR / YEAR) revert InvariantViolation();
+
+        ACCOUNTING_TOKEN.mintTo(STRATEGY, amount);
+    }
+
+    /**
+     * @notice Process losses by burning accounting tokens
+     * @param amount losses to burn
+     */
+    function processLosses(uint256 amount) external onlySafeManager checkAndResetCooldown {
+        uint256 totalSupply = ACCOUNTING_TOKEN.totalSupply();
+
+        // sanity check: if token.totalSupply() > small amount to prevent rounding issues
+        if (totalSupply < 1 ether) revert TvlTooLow();
+
+        // check lower bound - 10% of tvl (in bips)
+        if (amount > totalSupply * 1000 / DIVISOR) revert InvariantViolation();
+
+        ACCOUNTING_TOKEN.burnFrom(STRATEGY, amount);
+    }
+
+    /**
+     * @notice Set target APY to determine upper bound. e.g. 1000 = 10% APY
+     * @param targetApyInBips in bips
+     */
+    function setTargetApy(uint16 targetApyInBips) external onlySafeManager {
+        if (targetApyInBips > (DIVISOR / 2)) revert InvariantViolation();
+
+        emit TargetApyUpdated(targetApyInBips, targetApy);
+        targetApy = targetApyInBips;
+    }
+
+    /**
+     * @notice Set lower bound as a function of tvl for losses. e.g. 1000 = 10% of tvl
+     * @param lb in bips, as a function of % of tvl
+     */
+    function setLowerBound(uint16 lb) external onlySafeManager {
+        if (lb > (DIVISOR / 2)) revert InvariantViolation();
+
+        emit LowerBoundUpdated(lb, lowerBound);
+        lowerBound = lb;
+    }
+
+    /**
+     * @notice Set cooldown in seconds between every processing of rewards/losses
+     * @param cooldownSeconds_ new cooldown seconds
+     */
+    function setCoolDownSeconds(uint16 cooldownSeconds_) external onlySafeManager {
+        emit CooldownSecondsUpdated(cooldownSeconds_, cooldownSeconds);
+        cooldownSeconds = cooldownSeconds_;
+    }
+
+    /**
+     * @notice Allows an address with the SAFE_MANAGER_ROLE to specify a new safe address
+     * @param newSafe new address
+     */
+    function setSafeAddress(address newSafe) external virtual onlySafeManager {
+        emit SafeUpdated(newSafe, safe);
```

Comment:
good use of newSafe and safe values in event, we should have done this everywhere where we had an update